Volume/Supporting/Notifications.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  Notifications.swift
+//  Volume
+//
+//  Created by Hanzheng Li on 11/17/21.
+//  Copyright © 2021 Cornell AppDev. All rights reserved.
+//
+
+import AppDevAnalytics
+import Foundation
+import UserNotifications
+
+class Notifications: NSObject {
+    static let shared = Notifications()
+    let center = UNUserNotificationCenter.current()
+    
+    private override init() {
+        super.init()
+        center.delegate = self
+    }
+    
+    func requestAuthorization() {
+        center.getNotificationSettings { settings in
+            if settings.authorizationStatus == .notDetermined {
+                self.center.requestAuthorization(options: [.alert, .badge, .sound]) { granted, error in
+                    if granted {
+                        let event = VolumeEvent.enableNotification.toEvent(.notification, value: "", navigationSource: .unspecified)
+                        AppDevAnalytics.log(event)
+                    } else if let error = error {
+                        print("Error requesting push notification permissions: \(error)")
+                    }
+                }
+            }
+        }
+    }
+    
+    func registerForRemoteNotifications() {
+        if !UIApplication.shared.isRegisteredForRemoteNotifications {
+            DispatchQueue.main.async {
+                UIApplication.shared.registerForRemoteNotifications()
```

Comment:
The fact that we call this every time the app opens (even if the user declined push notifs) makes me slightly uncomfortable, but it's not a big deal. I also still don't understand how this works the first time you open the app. Wouldn't it be like this:

 1. Open app
 2. Check auth
 3. Request auth (popup)
 4. Try to register
 5. Fail registration 
 6. Approve auth (user clicks button)

Now what? Will you then try to register again, or do you need to wait until the next launch?